tests/runtime/usdt

```diff
@@ -1,5 +1,5 @@
 NAME "usdt probes - list probes by file"
-RUN bpftrace -l 'usdt:./testprogs/usdt_test'
+RUN bpftrace -l 'usdt:./testprogs/usdt_test:*'
```

Comment:
why is this needed?

Reply:
This is the normal syntax to "attach to all probes for that binary". Since I wanted to unify attachment and listing syntax, we either have to require the explicit wildcard for listing or make the wildcard implicit for attachment. I prefer the first option. Also, we would probably have to make the wildcard implicit for all probe types, which I don't really like.

Reply:
We try to avoid breaking compatibility with existing documentation (books) when possible.  But the current behaviour is inconsistent too:

```
$ sudo bpftrace_master  -l 'u:/bin/bash' | head -n2
uprobe:/bin/bash:__libc_csu_fini
uprobe:/bin/bash:__libc_csu_init
$ sudo bpftrace_master  -l 'k:' | head -n2
$
````

So I'm ok with breaking it as long as all probe types end up having consistent behaviour. @brendangregg  how do you feel about it.

But I do not like  the error this throws currently, its confusing:
```
$ sudo bpftrace -l 'u:/bin/bash'
stdin:1:13-14: ERROR: syntax error, unexpected {
u:/bin/bash {}
            ~
```

And can even segfault:

```
$ sudo bpftrace -l 'k:'
Segmentation fault
```

Reply:
> We try to avoid breaking compatibility with existing documentation (books) when possible.
> ...
> So I'm ok with breaking it as long as all probe types end up having consistent behaviour.

The current behaviour in this PR is that every probe part should be specified and non-empty (at least `*` should be given). The only part that can be empty/skipped is namespace for usdt (per documentation) and binary name for uprobe/usdt if PID is specified.

On a second thought, I would agree with empty part of a probe behaving as `*`. That would keep the backwards compatibility (at least for the above cases). Also, AFAIK, DTrace uses a similar syntax.

> 
> But I do not like the error this throws currently, its confusing:
> ...
> And can even segfault:

I'm currently working on this, this is why the PR is in draft state. I'm thinking about running one iteration of semantic analyser, which already has good error messages (and they would be consistent, too).